spec/unit/persistent_enum_spec.rb

```diff
@@ -463,19 +464,39 @@
   it "refuses to create a table in a transaction" do
     expect {
       ActiveRecord::Base.transaction do
-        create_test_model(:test_create_in_transaction, ->(t) { t.string :name }) do
+        create_test_model(:test_create_in_transaction, ->(t) { t.string :name; t.index [:name], unique: true }) do
           acts_as_enum([:A, :B])
         end
       end
     }.to raise_error(RuntimeError, /unsafe class initialization during/)
   end
 
+  it "refuses to create an enum without an index on the enum constant" do
+    expect {
+      ActiveRecord::Base.transaction do
+        create_test_model(:test_create_without_index, ->(t) { t.string :name }) do
+          acts_as_enum([:A, :B])
+        end
+      end
+    }.to raise_error(RuntimeError, /missing unique index/)
+  end
+
+  it "refuses to create an enum without a unique index on the enum constant" do
+    expect {
+      ActiveRecord::Base.transaction do
+        create_test_model(:test_create_without_index, ->(t) { t.string :name; t.index [:name] }) do
```

Comment:
`create_test_model` doesn't do cleanup if the class body throws, which is leaking this class.